### PR TITLE
test(ivy): add missing tests on directive lifecycle hooks

### PR DESCRIPTION
**Pull request by marclaval**
#22351 confirmed that these hooks should called for directives. This PR adds tests to show that.

### CODE REVIEW

File: packages/core/test/render3/lifecycle_spec.ts [hunk 1]
```diff
@@ -801,6 +777,45 @@ describe('lifecycles', () => {
...
+
+      it('should be called on directives after component', () => {
+        /** <comp directive></comp> */
+        function Template(ctx: any, cm: boolean) {
+          if (cm) {
+            elementStart(0, Comp, null, [Directive]);
+            elementEnd();
+          }
+          Comp.ngComponentDef.h(1, 0);
+          componentRefresh(1, 0);
```
Comment (kara) [inline packages/core/test/render3/lifecycle_spec.ts]: It looks like we're missing the directive's hostBindings and refresh here (it was probably missing from the original test too). Same on line 1268.

**Review by kara — COMMENTED** [review]
LGTM, one nit

File: packages/core/test/render3/lifecycle_spec.ts (hunk 1 above)
Comment (marclaval) [reply packages/core/test/render3/lifecycle_spec.ts]: Added. I also fixed the original test.

**Review by kara — APPROVED** [review]
LGTM

**Pull request closed without merging**